lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1032,12 +1272,37 @@ class BuildControl : public calyx::FuncOpPartialLoweringPattern {
         // Insert loop-latch at the end of the while group
         rewriter.setInsertionPointToEnd(whileBodyOpBlock);
         calyx::GroupOp whileLatchGroup =
-            getState<ComponentLoweringState>().getLoopLatchGroup(whileOp);
+            getState<ComponentLoweringState>().getWhileLoopLatchGroup(whileOp);
         rewriter.create<calyx::EnableOp>(whileLatchGroup.getLoc(),
                                          whileLatchGroup.getName());
 
         if (res.failed())
           return res;
+      } else if (auto *forSchedPtr = std::get_if<ForScheduleable>(&group);
+                 forSchedPtr) {
+        auto &forOp = forSchedPtr->forOp;
```

Comment:
Isn't this a mlir::Value, i.e., a pointer underneath? Do you need to take a reference?